hawc/apps/common/renderers.py

```diff
@@ -7,7 +7,27 @@
 from rest_framework.renderers import BaseRenderer
 from rest_framework.response import Response
 
-from .helper import FlatExport, rename_duplicate_columns
+from .helper import FlatExport, ReportExport, rename_duplicate_columns
+
+
+class DocxRenderer(BaseRenderer):
+    """
+    Renders a ReportExport object into a docx file.
+    """
+
+    media_type = "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+    format = ""
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        # throw error if we don't have a ReportExport
+        if not isinstance(data, ReportExport):
```

Comment:
i added a change upstream in #379 that allows us to short-circuit and return JSON if validation fails; consider bundling into this request (similar to how we handle the pandas renderer)